src/vivarium_testing_utils/automated_validation/data_loader.py

```diff
@@ -63,12 +65,12 @@ def get_dataset(self, dataset_key: str, source: DataSource) -> pd.DataFrame:
             self._add_to_cache(dataset_key, source, dataset)
             return dataset
 
-    def upload_custom_data(self, dataset_key: str, data: pd.DataFrame | pd.Series) -> None:
+    def upload_custom_data(self, dataset_key: str, data: pd.DataFrame) -> None:
         self._add_to_cache(dataset_key, DataSource.CUSTOM, data)
 
-    def _load_from_source(self, dataset_key: str, source: DataSource) -> None:
+    def _load_from_source(self, dataset_key: str, source: DataSource) -> pd.DataFrame:
         """Load the data from the given source via the loader mapping."""
-        return self._loader_mapping[source](dataset_key)
+        return self._loader_mapping[source](dataset_key)  # type: ignore[return-value]
```

Comment:
If you declare the type of `self._loader_mapping` in line 43 to be `dict[DataSource, Callable[[str], pd.DataFrame]]` you shouldn't need this ignore?